app/src/androidTest/java/org/mifos/mobile/ui/activities/LoginSetPasscodeTest.kt

```diff
@@ -0,0 +1,192 @@
+package org.mifos.mobile.ui.activities
+
+import android.view.View
+import android.view.ViewGroup
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.ViewInteraction
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.LargeTest
+import androidx.test.rule.ActivityTestRule
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.runner.AndroidJUnit4
+
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.TypeSafeMatcher
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mifos.mobile.R
+import org.mifos.mobile.utils.EspressoIdlingResouce
+
+/**
+ * UI Testing for Login using Espresso
+ *
+ * @author Ashwin Ramakrishnan
+ * @since 22/08/2020
+ */
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class LoginSetPasscodeTest {
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityTestRule(SplashActivity::class.java)
+
+    @Rule
+    @JvmField
+    val grantPermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(android.Manifest.permission.READ_PHONE_STATE)
+
+    @Before
+    fun registerIdlingResource() {
+        IdlingRegistry.getInstance().register(EspressoIdlingResouce.countingIdlingResource)
+    }
+
+    @After
+    fun unregisterIdlingResource() {
+        IdlingRegistry.getInstance().unregister(EspressoIdlingResouce.countingIdlingResource)
+    }
+
+    @Test
+    fun loginSetPasscodeTest() {
+        val usernameEditText = onView(
+            allOf(
+                childAtPosition(
+                    childAtPosition(
+                        withId(R.id.til_username),
+                        0
+                    ),
+                    0
+                )
+            )
+        )
+
+        val passwordEditText = onView(
```

Comment:
Let's follow convention, don't add a new line after every variable creation. Let's add lines as a delimiter for better readability, add them wisely.

Reply:
I'll look into those and remove them